### ISSUE TITLE
CentOS 7: yum isn't updating NodeJS

### ISSUE THREAD

**Issue by Jakobud** [nodesource/distributions#176]
Okay so here is my yum repo:

```
[nodesource]
name=Node.js Packages for Enterprise Linux 7 - $basearch
baseurl=https://rpm.nodesource.com/pub/el/7/$basearch
failovermethod=priority
enabled=1
gpgcheck=1
gpgkey=file:///etc/pki/rpm-gpg/NODESOURCE-GPG-SIGNING-KEY-EL

[nodesource-source]
name=Node.js for Enterprise Linux 7 - $basearch - Source
baseurl=https://rpm.nodesource.com/pub/el/7/SRPMS
failovermethod=priority
enabled=0
gpgkey=file:///etc/pki/rpm-gpg/NODESOURCE-GPG-SIGNING-KEY-EL
gpgcheck=1
```

Here is what I have installed:
```
]# yum info nodejs
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirror.evolvedservers.com
 * epel: mirrors.tummy.com
 * extras: centos.arvixe.com
 * remi-safe: mirrors.mediatemple.net
 * updates: mirror.thelinuxfix.com
Installed Packages
Name        : nodejs
Arch        : x86_64
Version     : 4.1.0
Release     : 1nodesource.el7.centos
Size        : 23 M
Repo        : installed
Summary     : JavaScript runtime
URL         : http://nodejs.org
License     : MIT and ASL 2.0 and ISC and BSD
Description : Node.js is a platform built on Chrome\'s JavaScript runtime
            : for easily building fast, scalable network applications.
            : Node.js uses an event-driven, non-blocking I/O model that
            : makes it lightweight and efficient, perfect for data-intensive
            : real-time applications that run across distributed devices.
```

So I want to update to 5.x today. I simply get this:

```
# yum update nodejs
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirror.evolvedservers.com
 * epel: mirrors.tummy.com
 * extras: centos.arvixe.com
 * remi-safe: mirrors.mediatemple.net
 * updates: mirror.thelinuxfix.com
No packages marked for update
```

What am I not understanding here? I see Node 5.x versions here:

https://rpm.nodesource.com/pub_5.x/el/7/x86_64/

Why is yum not automatically updating to the latest version? Am I just misunderstanding how to use yum for updating to a particular version?

This also didn't work:

```
# yum --disablerepo=* --enablerepo=nodesource update nodejs
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
No packages marked for update
```

Also
```
# yum --disablerepo=* --enablerepo=nodesource list available nodejs
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
Error: No matching Packages to list
```

**Comment by chrislea** [comment]
We keep 4.2.x (LTS) and 5.x (new hotness) in separate repos so people don't get forcibly updated without meaning to. If you run the 5.x setup script, it should update things as needed to pick up the 5.x branch. Instructions are [here](https://github.com/nodesource/distributions/#rpminstall).

**Comment by Jakobud** [comment]
Okay, so the version I have now is 4.1.0. How come the repo isn't at least offering me 4.2.x?

If I run the 5.x setup bash script will that great a separate repo entry that i can install 5.x from?

On a related note, how come you guys don't just make a .repo file that has all the different repositories in it and allow people to simply pick and choose if they want to `--enablerepo=nodesource5` or `--enablerepo=nodesource4` etc? Seems much more straightforward. I dunno :-)

**Comment by Jakobud** [comment]
Okay.... still not working. Ran this successfully:

```
curl -sL https://rpm.nodesource.com/setup_5.x | bash -
```

Now here is my repo:
```
[nodesource]
name=Node.js Packages for Enterprise Linux 7 - $basearch
baseurl=https://rpm.nodesource.com/pub_5.x/el/7/$basearch
failovermethod=priority
enabled=1
gpgcheck=1
gpgkey=file:///etc/pki/rpm-gpg/NODESOURCE-GPG-SIGNING-KEY-EL

[nodesource-source]
name=Node.js for Enterprise Linux 7 - $basearch - Source
baseurl=https://rpm.nodesource.com/pub_5.x/el/7/SRPMS
failovermethod=priority
enabled=0
gpgkey=file:///etc/pki/rpm-gpg/NODESOURCE-GPG-SIGNING-KEY-EL
gpgcheck=1
```

```
# yum --disablerepo=* --enablerepo=nodesource list available
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
Available Packages
nodejs-debuginfo.x86_64                               0.10.40-1nodesource.el7.centos                               nodesource
nodejs-devel.x86_64                                   0.10.40-1nodesource.el7.centos                               nodesource
nodejs-docs.noarch                                    0.10.40-1nodesource.el7.centos                               nodesource
```

Node 0.10.40 ?????

**Comment by chrislea** [comment]
I'm not sure how this was initially configured for you so I'm not sure why it's not pushing you to 4.2.x. But if you use the `setup_4.x` script it should set that up. All the `setup_X.X` scripts do is figure out what your distro is, and install a corresponding `nodesource-release-*.rpm` which is what actually puts all the repo stuff in place.

We keep the repos totally separate just for clarity. Most people just want to run a setup script of some sort once and then not have to touch anything again for a long time.

Hope this helps!

**Comment by Jakobud** [comment]
Thanks. Did you see my above reply about the 5.x repo offering me Node 0.10.40? Something weird going on...

**Comment by chrislea** [comment]
Yes. I just tried spinning up a fresh EL7 box and running the same install script to make sure nothing super weird is going on (it's not, it worked for me). So without having access to your machine and time to remember a bunch of stuff I might need to brush up on regarding `yum` and `rpm` I'm not exactly sure what you are seeing.

The repo file looks correct to me. I'd try clearing the yum cache, running `yum update`, and then seeing what shows up.

**Comment by Jakobud** [comment]
Okay I got it to work. I think the thing that threw me off was `yum list available` for the nodesource repo was returning the node version as `0.10.40`, but when I attempted to `yum install nodejs` it offered up 5.1.x to install. Weird. Here is my terminal:

```
]# cat nodesource-el.repo
[nodesource]
name=Node.js Packages for Enterprise Linux 7 - $basearch
baseurl=https://rpm.nodesource.com/pub_5.x/el/7/$basearch
failovermethod=priority
enabled=1
gpgcheck=1
gpgkey=file:///etc/pki/rpm-gpg/NODESOURCE-GPG-SIGNING-KEY-EL

[nodesource-source]
name=Node.js for Enterprise Linux 7 - $basearch - Source
baseurl=https://rpm.nodesource.com/pub_5.x/el/7/SRPMS
failovermethod=priority
enabled=0
gpgkey=file:///etc/pki/rpm-gpg/NODESOURCE-GPG-SIGNING-KEY-EL
gpgcheck=1
[root@dev01 yum.repos.d]# yum --disablerepo=* --enablerepo=nodesource search nodejs
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
======================================================================================= N/S matched: nodejs ========================================================================================
nodejs-debuginfo.x86_64 : Debug information for package nodejs
nodejs.x86_64 : JavaScript runtime
nodejs-devel.x86_64 : JavaScript runtime - development headers
nodejs-docs.noarch : Node.js API documentation

  Name and summary matches only, use "search all" for everything.
[root@dev01 yum.repos.d]# yum --disablerepo=* --enablerepo=nodesource list available nodejs
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
Available Packages
nodejs.x86_64                                                                       0.10.40-1nodesource.el7.centos                                                                        nodesource
[root@dev01 yum.repos.d]# yum --disablerepo=* --enablerepo=nodesource install nodejs
Loaded plugins: fastestmirror
nodesource                                                                                                                                                                   | 2.5 kB  00:00:00
nodesource/x86_64/primary_db                                                                                                                                                 |  10 kB  00:00:08
Loading mirror speeds from cached hostfile
Resolving Dependencies
--> Running transaction check
---> Package nodejs.x86_64 0:5.1.0-2nodesource.el7.centos will be installed
--> Finished Dependency Resolution

Dependencies Resolved

===============================================================================
 Package                                 Arch                                    Version                                                          Repository                                   Size
===============================================================================
Installing:
 nodejs                                  x86_64                                  5.1.0-2nodesource.el7.centos                                     nodesource                                  8.1 M

Transaction Summary
===============================================================================
Install  1 Package

Total download size: 8.1 M
Installed size: 28 M
Is this ok [y/d/N]:
```

Strange. But `install` worked.

**Comment by Kmumuda** [comment]
Available: 2:nodejs-devel-12.22.8-1nodesource.x86_64 (nodesource)
               nodejs-devel(x86-64) = 2:12.22.8-1nodesource
           Available: 2:nodejs-devel-12.22.9-1nodesource.x86_64 (nodesource)
               nodejs-devel(x86-64) = 2:12.22.9-1nodesource
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
Is this issues resolved?
can i get any solution for this
